### PR TITLE
Edits to utilsFiles to facilitate searching for / replacing text strings in an entire repository, and easily retrieving the list of locations

### PR DESCRIPTION
**Pull request by epi-jhd**
- Add output to txtFindReplace, xlsxFindReplace, fileFindReplace and FilesFindReplace to list the filenames = number of changes

- Add function getAllFilepaths which finds all files of a specified type within a directory.

These changes are designed to facilitate the work of the admin team when they change the drive file structure and we all need to hunt through our scripts to find if/where we have referenced files on the drive and change them to match the new path.

### CODE REVIEW

File: R/utilsFiles.R
Comment (EPI-JHS) [inline R/utilsFiles.R:1]: Tested find and replace - 10/10!

File: R/utilsFiles.R [hunk 1]
```diff
@@ -376,10 +385,49 @@ xlsxFindReplace <-
...
+#' @param ignore.these The names of any folders you wish to ignore.
 
+#'
+#' @return Returns list of complete filepaths
+#' @export
+#' @examples
+#' \dontrun{
+#' allpaths <- getAllFilepaths(getwd(),filetypes = c("R","Rmd","txt"), ignore.these = c(".git", ".gitignore",".Rproj.user"))
+#' }
+getAllFilepaths <- function(directory, filetypes = NULL, ignore.these=NULL){
```
Comment (EPI-JHS) [inline R/utilsFiles.R:411]: Also works great for me in my tested directly. Are we intending to combine this with find and replace to bulk change any mention?

**Review by EPI-JHS — COMMENTED** [review]
Tested find and replace + get all file paths, both work great

File: R/utilsFiles.R (hunk 1 above)
Comment (epi-jhd) [reply R/utilsFiles.R:411]: For finding locations for / making bulk changes, this combo works to get a data frame that can then query to identify all the files with the old filepath segment (and change at same time if needed).

allfiles <- getAllFilepaths(getwd(),filetypes = c("R","Rmd","txt"), ignore.these = c(".git", ".gitignore",".Rproj.user"))
test <- filesFindReplace(allfiles ,"[oldfilepathsegment]","[newfilepathsegment]", listonly = T)
res <- data.frame("filename" = names(test)
                  ,"changes" = test)

Comment (epi-jhd) [reply R/utilsFiles.R:411]: also thanks for reviewing :D

**Comment by gdesve** [comment]
Thanks Jenny, 
I will work on STRAP this week and that review will be part of my duty ;)

File: R/utilsFiles.R (hunk 1 above)
Comment (epi-gde) [reply R/utilsFiles.R:411]: You should manage to never have path in your source ;) 
But this is probably for old sources ;)

**Comment by epi-gde** [comment]
Thanks for changes on txtFindReplace, xlsxFindReplace, fileFindReplace and FilesFindReplace , listing number of changes is great  and work well !

**Comment by epi-gde** [comment]
I see some redondance between listFiles and getAllFilepath, why not using the first one ?

**Comment by epi-jhd** [comment]
As far as I understand, listFiles looks into just one level (ie the folder its given) - I needed a function that follows each filepath to the end and can thus get ALL the files within every folder and sub folder. 
Perhaps we could edit listFiles to have this option? as well as the option to show the full filepath, not just the the file name.

**Comment by epi-gde** [comment]
> As far as I understand, listFiles looks into just one level (ie the folder its given) - I needed a function that follows each filepath to the end and can thus get ALL the files within every folder and sub folder. Perhaps we could edit listFiles to have this option? as well as the option to show the full filepath, not just the the file name.

ListFiles look recursively into each subdirectories and all sub sub directories (GetAllFiles only looks in first level of subdirectories) . ListFiles has an option to forget hiden directories but no easy option to exclude some 
ListFiles has a pattern option which let you choose all files with vaccin in name but no option to easily select types (even if this can be done with pattern) 
Then I suggest to merge them by integrating getAllFiles option into listFiles (which is natively recursive by usin base R functions)

**Comment by epi-jhd** [comment]
I found the recusive functionality of listFiles (did not know abou this!) and it thus does everything required. 
I have remove getAllFilespaths as not needed.

**Comment by epi-jhd** [comment]
Ready to merge?

**Pull request merged**